src/ralph/data_importer/tests/test_fields.py

```diff
@@ -0,0 +1,38 @@
+from django.contrib.auth import get_user_model
+from django.test import TestCase
+
+from ralph.accounts.tests.factories import UserFactory
+from ralph.data_importer.fields import ThroughField
+from ralph.data_importer.resources import LicenceResource
+from ralph.data_importer.widgets import LicenceUserWidget
+from ralph.licences.models import Licence, LicenceUser
+from ralph.licences.tests.factories import LicenceFactory
+
+
+class DataImporterFieldsTestCase(TestCase):
+    def setUp(self):  # noqa
+        self.licence = LicenceFactory()
+        self.users = [UserFactory() for i in range(4)]
+        LicenceUser.objects.create(
+            licence=self.licence, user=self.users[0]
+        )
+
+    def test_through_field(self):
+        field = ThroughField(
+            through_model=LicenceUser,
+            through_from_field_name='licence',
+            through_to_field_name='user',
+            attribute='users',
+            column_name='users',
+            widget=LicenceUserWidget(model=get_user_model())
+        )
+
+        self.assertEqual(self.licence.users.all().count(), 1)
+
+        with self.assertNumQueries(4):
```

Comment:
imo there should be max 4 queries:
* `self.widget.clean(value)`
* `self.through_model.objects.all()`
* `self.through_model.objects.bulk_create(to_add_list)`
* `self.through_model.objects.filter(...).delete()`

but in your case there is any remove

Could you remove one `LicenceUser` in this test too?